### DISCUSSION TITLE
How to install RecBole on windows PyCharm? (package versions have conflicting dependencies error)

### DISCUSSION THREAD

**Discussion by alanjacob10** [RUCAIBox/RecBole#806] [Q&A]
Hi,

Does anybody know why i get the following error when trying to install recbole in PyCharm? :

-

Error occured:
`ERROR: Cannot install recbole==0.1.0, recbole==0.1.1, recbole==0.1.2 and recbole==0.2.1 because these package versions have conflicting dependencies.

Proposed solution:
Try to run this command from the system terminal. Make sure that you use the correct version of 'pip' installed for your Python interpreter located at 'C:\Users\AlanR\PycharmProjects\recboletest1\venv\Scripts\python.exe'.

Command Output:
Collecting recbole
  Using cached recbole-0.2.1-py3-none-any.whl (2.0 MB)
Collecting colorlog==4.7.2
  Using cached colorlog-4.7.2-py2.py3-none-any.whl (10 kB)
Collecting pyyaml>=5.1.0
  Using cached PyYAML-5.4.1-cp37-cp37m-win32.whl (193 kB)
Collecting colorama==0.4.4
  Using cached colorama-0.4.4-py2.py3-none-any.whl (16 kB)
Collecting scikit-learn>=0.23.2
  Using cached scikit_learn-0.24.1-cp37-cp37m-win32.whl (6.1 MB)
Collecting recbole
  Using cached recbole-0.1.2-py3-none-any.whl (2.0 MB)
  Using cached recbole-0.1.1-py3-none-any.whl (2.0 MB)
  Using cached recbole-0.1.0-py3-none-any.whl (2.0 MB)
Collecting hyperopt>=0.2.4
  Using cached hyperopt-0.2.5-py2.py3-none-any.whl (965 kB)

The conflict is caused by:
    recbole 0.2.1 depends on torch>=1.7.0
    recbole 0.1.2 depends on torch>=1.6.0
    recbole 0.1.1 depends on torch>=1.6.0
    recbole 0.1.0 depends on torch>=1.6.0

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: Cannot install recbole==0.1.0, recbole==0.1.1, recbole==0.1.2 and recbole==0.2.1 because these package versions have conflicting dependencies.
ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/user_guide/#fixing-conflicting-dependencies`

-

I tried updating pip, and also install recbole through commando promt but didn't seem to react to this. Still can't install in PyCharm.

Ps. Recbole support kindly helped me with another issue in another discussion, but do to too much memory power required by the algorithms I had to switch to a windows computer with more RAM and here this issue occured.

Sincerely,
Alan

**Comment by hyp1231** [comment]
It seems that you have pytorch < 1.6.0 on your Windows device, while RecBole 0.2.1 requires pytorch >= 1.7.0.

Upgrade your pytorch package and try again?

**Reply by alanjacob10** [reply]
Yes I tried this... 
It gave the following error when trying to install through shell (didn't work in PyCcharm):
![Skærmbillede 2021-04-12 100956](https://user-images.githubusercontent.com/81938012/114362409-96e5fb00-9b77-11eb-9991-bf121734d3f8.png)

And apparently the latest version was already found on my windows.:
![Skærmbillede 2021-04-12 101115](https://user-images.githubusercontent.com/81938012/114362494-aebd7f00-9b77-11eb-9adc-5ab330a48cf6.png)

**Reply by hyp1231** [reply]
I cannot make sure if you install PyTorch on the correct environment...

Maybe you can open your PyCharm project, create a new .py file, run the following code
```python
import torch
print(torch.__version__)
```
to check the PyTorch version on your PyCharm environment.

**Reply by alanjacob10** [reply]
What do you mean with installing to the correct enviroment? how do you do this?

I tried what you wrote but it didn't find any torch version.


Regards,
Alan

**Reply by hyp1231** [reply]
Hi Alan,

Maybe you can learn about how to [configure a Python interpreter in a PyCharm project](https://www.jetbrains.com/help/pycharm/configuring-python-interpreter.html) and [install, uninstall, and upgrade packages  in a PyCharm project](https://www.jetbrains.com/help/pycharm/installing-uninstalling-and-upgrading-packages.html) firstly before installing recbole.
